src/photon.cpp

```diff
@@ -156,8 +156,14 @@ PhotonInteraction::PhotonInteraction(hid_t group)
 
     // Read binding energy and number of electrons
     hid_t tgroup = open_group(rgroup, designator.c_str());
-    read_attribute(tgroup, "binding_energy", shell.binding_energy);
-    read_attribute(tgroup, "num_electrons", shell.n_electrons);
+
+    // Read binding energy energy and number of electrons if atomic relaxation
+    // data is present
+    if (attribute_exists(tgroup, "binding_energy")) {
+      has_atomic_relaxation_ = true;
+      read_attribute(tgroup, "binding_energy", shell.binding_energy);
```

Comment:
The subshell binding energy is also used for the photoelectric effect -- any idea what we should do there if the binding energy is missing?

Out of curiosity, is the subshell transition data also missing in these libraries?

Reply:
Very good point. I've just updated the branch to account for the possibly-missing binding energy when handling photoelectric. An easy fallback in that case is to use the binding energies from the Compton profile data, which will always be present.

The subshell transition data effectively _is_ the atomic relaxation data. It just so happens that we write the binding energies / number of electrons as part of the atomic relaxation data, so one could check either as a way of setting `has_atomic_relaxation_`, but this block here is the first place we run into it.